tests/unit/Registry.ts

```diff
@@ -97,57 +96,54 @@ registerSuite({
 				assert.isNull(item);
 			},
 			'replaces promise with result on resolution'() {
-				let resolveFunction: any;
 				const promise: any = new Promise((resolve) => {
-					resolveFunction = resolve;
+					setTimeout(() => {
```

Comment:
Is a timeout better than resolving explicitly?

Reply:
We had a difficult time getting the registry tests to work properly.  The setTimeout calls were an experiment that now we don't need so I will remove them.